### PR TITLE
add polarionID for test verify hook annotation updates domainxml

### PR DESCRIPTION
**Pull request by Ahmad-Hafe**
### What this PR does
Before this PR:
no polarion id for  update domain XML with SM BIOS properties

After this PR:
adding polarion id for  update domain XML with SM BIOS properties

Fixes #

### Why we need it and why it was done in this way
polarion ID needed so test can be wxecutable in Tire1

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note

```

### CODE REVIEW

File: tests/vmi_hook_sidecar_test.go [hunk 1]
```diff
@@ -293,7 +293,7 @@ var _ = Describe("[sig-compute]HookSidecars", decorators.SigCompute, func() {
 
 		Context("with ConfigMap in sidecar hook annotation", func() {
 
-			DescribeTable("should update domain XML with SM BIOS properties", func(withImage bool) {
+			It("[test_id:10907] should update domain XML with SM BIOS properties", func(withImage bool) {
```
Comment (tiraboschi) [inline tests/vmi_hook_sidecar_test.go:296]: `DescribeTable` is used here to run a table driven set of tests.
You cannot simply mark it as a scalar node with `It`.

**Review by EdDev — CHANGES_REQUESTED** [review]
Kubevirt is not aware of polarion or uses it itself. Please try to reason the need for this without involving some d/s usage of it.

File: tests/vmi_hook_sidecar_test.go (hunk 1 above)
Comment (Ahmad-Hafe) [reply tests/vmi_hook_sidecar_test.go]: fixed now
I just added the polarionID to DescribeTable

**Comment by Ahmad-Hafe** [comment]
@tiraboschi can you please review

**Review by tiraboschi — APPROVED** [review]

**Comment by Ahmad-Hafe** [comment]
@EdDev can you please review?

File: tests/vmi_hook_sidecar_test.go [hunk 2]
```diff
@@ -303,17 +303,24 @@ var _ = Describe("[sig-compute]HookSidecars", decorators.SigCompute, func() {
 
 		Context("with ConfigMap in sidecar hook annotation", func() {
 
-			It("should update domain XML with SM BIOS properties", func() {
+			DescribeTable("[test_id:9833]should update domain XML with SM BIOS properties", func(withImage bool) {
 				cm, err := virtClient.CoreV1().ConfigMaps(testsuite.GetTestNamespace(vmi)).Create(context.TODO(), RenderConfigMap(), metav1.CreateOptions{})
 				Expect(err).ToNot(HaveOccurred())
-				vmi.ObjectMeta.Annotations = RenderSidecarWithConfigMap(hooksv1alpha2.Version, cm.Name)
+				if withImage {
```
Comment (EdDev) [inline tests/vmi_hook_sidecar_test.go:299]: Using a boolean like this is not a good practice.
Booleans as arguments to a function/method are a smell, although sometimes it makes sense.

Here it does not make sense and you better pass the render function as input instead.
```go
DescribeTable("[test_id:9833]should update domain XML with SM BIOS properties", func(generateAnnotation func(string, string) bool) {
  ...
  vmi.ObjectMeta.Annotations = generateAnnotation(hooksv1alpha2.Version, cm.Name)
  ...
},
  Entry("when sidecar image is specified", RenderSidecarWithConfigMapPlusImage),
  Entry("when sidecar image is not specified", RenderSidecarWithConfigMapWithoutImage),
)
```

File: tests/vmi_hook_sidecar_test.go [hunk 3]
```diff
@@ -303,17 +303,24 @@ var _ = Describe("[sig-compute]HookSidecars", decorators.SigCompute, func() {
 
 		Context("with ConfigMap in sidecar hook annotation", func() {
 
-			It("should update domain XML with SM BIOS properties", func() {
+			DescribeTable("[test_id:9833]should update domain XML with SM BIOS properties", func(withImage bool) {
```
Comment (EdDev) [inline tests/vmi_hook_sidecar_test.go:296]: The commit is not reflecting that you added a new test. please refelct it in the commit or split the two and reason for each.

**Review by EdDev — CHANGES_REQUESTED** [review]
In addition to the inline comments, here are some general issues:
- The 2nd commit is a merge one, which should not be seen here. You are suppose to rebase over the current `main` branch locally and then push the changes. Such commits should not be seen in the PR.
- As I commented earlier in this PR, you cannot use "polarion" or any other tool for a reason to add this ID. You may use this tool on D/S, but this change is proposed on U/S.
Therefore, I suggest removing any mentioning of "polarion" and add a better reason why the ID is needed/useful.

**Comment by Ahmad-Hafe** [comment]
created different PR https://github.com/kubevirt/kubevirt/pull/12013

**Pull request closed without merging**